### PR TITLE
SLEP024 Guideline for external posts on scikit-learn blog

### PR DESCRIPTION
**Pull request by glemaitre**
This SLEP intends to have a discussion to define a clear guideline allowing to accept external contribution in the scikit-learn blog post.

### CODE REVIEW

File: slep024/proposal.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Detailed description
+--------------------
+
+Scikit-learn has a blog post available at the following URL:
+https://blog.scikit-learn.org/. Since its origin, the blog post is used to relay
+information related to diverse subject such as sprints, interviews of contributors,
+collaborations, and technical content.
+
+When it comes to technical content, up to now, the content is only limited to the
+scikit-learn library. However, the scikit-learn community is going beyond the
```
Comment (jjerphan) [inline slep024/proposal.rst:27]: Thank you for opening the discussion.

If I can interject, some content is also exposed on external website but only linked on the blog post. This is for instance the case of [the series on performance improvements](https://blog.scikit-learn.org/technical/performances/).

I think it would be better to have this part of scikit-learn's blog post directly.

This was not done originally because of lack of time and for convenience (it was easy for me to publish it on my website and to iterate on it), but I do no mean to be the sole owner on the knowledge shared there.

Should such external content be discussed as part of this SLEP?

Comment (glemaitre) [reply slep024/proposal.rst:27]: You are faster than me writing the SLEP :).

For this particular case, this is not even a question since this already linked to some internal of scikit-learn. So it would go to facto in the scikit-learn blog if you ask me.

But let's imagine that this is a topic that is related to scikit-learn but somehow outside of the library itself. Then, I would consider this case as part of the SLEP. The guidelines should answer to some questions with some extend, notably if this is eligible for inclusion.

Note, that my first thought here with this SLEP was more on: someone has a shiny compatible package and search some visibility; is it possible to advertise it and if so, what are the couple of requirements from our side.

Comment (jjerphan) [reply slep024/proposal.rst:27]: OK, I was unsure my remark was relevant with regards to the subject of this SLEP (and it is not based on your remark).

Should I open an issue or PRs directly to integrate its content in scikit-learn's blog?

Comment (glemaitre) [reply slep024/proposal.rst:27]: Yep I would find it relevant.

Comment (jjerphan) [reply slep024/proposal.rst:27]: I just have opened https://github.com/scikit-learn/blog/pull/191.

File: slep024/proposal.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+To accept an external contribution, the blog post should be related to the scikit-learn
+ecosystem. When it comes to presenting a compatible tool, the criteria are the
+following:
+
+- The tool should be compatible with scikit-learn.
+- The tool should be under an open-source license.
+- The tool should be actively maintained.
+- The tool should have a clear documentation.
+- The tool should be well tested.
+- The tool should not be a commercial product or serve advertisement for a company.
```
Comment (adrinjalali) [inline slep024/proposal.rst:59]: This is very similar to what we had when I was in NumFocus's affiliated project committee. These criteria are quite hard to assess, and also quite hard to maintain. They seem like a low bar, but they're indeed quite high of a passing bar.

I would probably modify this to something like:

The scikit-learn blog is not an opinionated place when it comes to tools. Posts are included if somebody takes the effort of writing them. However, we don't want the blog to be a place where it's flooded by companies trying to advertise their products. Therefore we have the following requirements:

- The post needs to cover the basic theoretical background required to understand the post. Our blog posts are educational resources.
- The tools used in the blog need to be under an OSI approved or similar license.
- The content of the post should be limited to the tool at hand, rather than using the tool to advertise a commercial ecosystem / tool.
- The proposing community needs to adhere to our code of conduct standards. We reserve the rights to reject posts from communities where there are issues that we'd like to stay away from.

Comment (betatim) [reply slep024/proposal.rst:59]: The current content focusses a lot on the tool/library that the blog post is covering. Adrin's suggestion is more about the content of the blog post. I think that is the right direction to go. I've been thinking about how to express my thoughts and so far the best I've come up with is that we should try and highlight what we do want to see in the blog posts (as opposed to focussing on what we don't want).

What do we want? My thoughts: the blog, like the rest of the documentation, should aim to be recognised as authoritative and high quality. Better to not write a blog post if it would be just average or a repeat of what can be read elsewhere. The content should be obviously correct. By this I mean a statement should be easy to read, easy to understand and easy to come to the conclusion of the statement. As opposed to statements which are hard to parse, might make you conclude A when in reality you should be concluding B and the author really rather you conclude A. I guess a phrase for this kind of language is "technically correct, but misleading".

Like scikit-learn the blog should cover things which are well established and "old". They might be less well known or been forgotten, but it shouldn't be "trail blazing".

Like a wikipedia article there should be sources you can link to to back up your claims and give the reader more details, etc. For me it would be fine to link to code that you can run as "sources" for say benchmarks. You wouldn't have to publish your results elsewhere first.

You should stick to the spirit of these guidelines, sticking to the "letter of the law" is not enough. This means there will be an element of human judgement.

Clearly mark "paid content"? I am a bit less sure about this one. I know in a lot of newspapers, magazines, youtube, instagram, etc content people are obviously and not so obviously rewarded for talking about something. This can be a ski resort inviting a reporter and paying for the trip, with the hope that they write about their time in the resort, a free pizza oven sent to a cooking YouTuber, straight up paid for advertising in a newspaper. I don't think there is a fundamental problem with this kind of "paid content" (using a broad definition here). And you could argue that it is covered under "take care that the reader understands" from above. Depending on how big the influence is something very prominent like "I work for company X and this post is about things we make, so take what I say with a grain of salt" or a note at the end "Project Y invited and paid for me to travel to their sprint" is appropriate. As long as the facts are with you, there seems to be very little downside to declaring your possible conflicts. It might even make you more credible.

Comment (ogrisel) [reply slep024/proposal.rst:59]: I would be fine in giving organizations that financially sponsor the project by hiring (near-)full-time core contributors to use the scikit-learn blog to advertise open source resources related to scikit-learn in their own name as long as the relationship to the scikit-learn project is made explicit in the blog post. One recent case is:

https://github.com/scikit-learn/blog/pull/217

Note that in this case, this is not a tool but an educational platform. In this case, the material is published on a GitHub repository with open license, but this is not a software library/application so it's a bit off w.r.t. the current SLEP but maybe this is fine.

**Comment by GaelVaroquaux** [comment]
Hi there,
I feel that we need to make scikit-learn's communication for active (lately we have all been overwhelmed and thus not communicating much), so I'm looking at the various TODOs around communication.
I came upon this, and I feel that communication guidelines are particularly important if we want to onboard new communication person. However, I wonder if a SLEP is the right way to draft them, and even to host them, as our goal with the comms team is to enable a different type of people, with more comms DNA than developer DNA.
What do people think? Should we continue here, or draft a file in the comms repo https://github.com/scikit-learn/communication/ ?

**Comment by GaelVaroquaux** [comment]
Digging a bit more in the comm's organization, I found that we have a backlog of posts that got nowhere despite looking quite aligned with what we want in terms of communication. For instance: https://github.com/scikit-learn/blog/pull/192

This comforts my feeling that we want comms to grew as a team that feels enabled and free to move without unnecessary restrictions, and moving these directives from SLEP to a comms repo would help.

**Comment by GaelVaroquaux** [comment]
Given that this SLEP was both fairly advanced and had given rise to interesting discussion, I chose to integrate the feedback from the discussion into the SLEP.
The SLEP process is interesting for consensus creation.

I'm thinking of calling a vote on this soonish

**Comment by ylechelle** [comment]
Please allow me to propose additional food for thoughts. As some of you know, I am driving the development of [Scikit-learn Central](https://scikit-learn-central.probabl.ai) from within [Probabl](https://probabl.ai). I would love to be able to contribute content for the scikit-learn blog itself, especially as we are considering featuring libraries that make it to the catalog, as well as some use cases, do give them more depth and trigger conversations. We would produce content as Probabl, but a lot of it would also make a lot of sense reposted and/or retrofitted to feed the scikit-learn blog, of course, validated by the collective as usual. Happy to brainstorm some more about this.

File: slep024/proposal.rst [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+criteria.
+
+Commercial links
+^^^^^^^^^^^^^^^^
+
+A blog article  should not be on a commercial product or serve advertisement for a
+company. Likewise the content of the post should be limited to the tool at hand, rather
+than using the tool to advertise a commercial ecosystem / tool.
+
+However, organizations that financially sponsor the project by hiring (near-)full-time
```
Comment (thomasjpfan) [inline slep024/proposal.rst:90]: Does this only apply to organizations that hire core contributors or any organization that provides financial support to scikit-learn? For example, if an organization were to provide a venue for a scikit-learn sprint, can they write a blog post on an open source tool they developed and end with "if you are interested in this work, they are hiring"?

**Review by thomasjpfan — COMMENTED** [review]
I'm +1 on these guidelines for external posts.

File: slep024/proposal.rst [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+- Claims should, as much as possible, backed by authoritative sources or reproducible
+  code (see below)
+
+Note that there will be an element of human judgment in applying these inclusion
+criteria.
+
+Commercial links
+^^^^^^^^^^^^^^^^
+
+A blog article  should not be on a commercial product or serve advertisement for a
```
Comment (glemaitre) [inline slep024/proposal.rst]:
```suggestion
A blog article should not be on a commercial product or serve advertisement for a
```

File: slep024/proposal.rst (hunk 3 above)
Comment (GaelVaroquaux) [reply slep024/proposal.rst:90]: I think that your example is a good example of a legitimate plug.

In my mind, a legitimate plug would be "proportional" to the service to the project. That's really hard to put in wording. Happy to hear if anyone has a suggestion

**Comment by betatim** [comment]
To reiterate my position (taking from GitHub Universe submission guide): Kick the sales pitch - Sessions are intended to be educational, not salesy. When crafting your abstract, focus on sharing actionable insights, real-world examples, and best practices that deliver value to attendees. Avoid heavy product promotion, and instead engage your audience with ideas and lessons they can apply.

As long as we get that right for the blog I think we are a long way towards having a good blog.
